linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -942,6 +969,16 @@ private final class Analyzer(config: CommonPhaseConfig,
       val subclasses = instantiatedSubclasses
       for (subclass <- subclasses)
         subclass.callMethodResolved(methodName)
+
+      if (checkAbstractReachability) {
+        /* Also lookup the method as abstract from this class, to make sure it
+         * is *declared* on this type. We do this after the concrete lookup to
+         * avoid work, since a concretely reachable method is already marked as
+         * abstractly reachable.
+         */
+        if (!methodName.isReflectiveProxy)
+          lookupAbstractMethod(methodName).reachAbstract()
```

Comment:
Hum, don't we have to reach on *all* abstract definitions? Otherwise the value of `isAbstractReachable` will be non-deterministic (or at least dependent on the order of interfaces).

Reply:
Note: I'm OK with simply not exposing `isAbstractReachable` in which case this would be fine IIUC.

Reply:
No, we don't have to reach all abstract definitions. We just need to make sure that one is reachable. The choice is actually deterministic: it is a function of the superclasses and the interfaces that they implement. Note that the interface list is ordered, so this is deterministic.

Hopefully the test I added will make it obvious.